File: HW14Assigned/low8/duck/DecoyDuck.java
package low8.duck;
import low8.fly.*;
import low8.quack.*;
import low8.eggs.*;

public class DecoyDuck extends Duck {

   public DecoyDuck( ) {
      flyBehavior = new FlyNoWay( );
      quackBehavior = new MuteQuack( );
      laysEggsBehavior = new DoesNotLayEggs( );
   }

   public void display( ) {
      System.out.println("I’m a decoy duck");
   }
}

File: HW9Assigned/low8/Unique/DecoyDuck.cpp
#include <iostream>
#include "DecoyDuck.h"
#include "FlyNoWay.h"
#include "MuteQuack.h"

DecoyDuck::DecoyDuck( ) {
   flyBehavior = new FlyNoWay( );
   quackBehavior = new MuteQuack( );
}

void DecoyDuck::display( ) {
   std::cout << "I’m a decoy duck" << std::endl;
}
